vm/actor/src/builtin/market/mod.rs

```diff
@@ -4,7 +4,687 @@
 pub use self::state::State;
+pub use self::types::*;
+use crate::{
+    make_map, request_miner_control_addrs, BalanceTable, DealID, DealWeight, Multimap,
+    OptionalEpoch, SetMultimap, BURNT_FUNDS_ACTOR_ADDR, CALLER_TYPES_SIGNABLE, MINER_ACTOR_CODE_ID,
+    SYSTEM_ACTOR_ADDR,
+};
+use address::Address;
+use cid::Cid;
+use clock::ChainEpoch;
+use encoding::to_vec;
+use ipld_amt::Amt;
+use ipld_blockstore::BlockStore;
+use message::Message;
+use num_bigint::bigint_ser::BigIntSer;
+use num_bigint::{BigInt, BigUint};
+use num_derive::FromPrimitive;
+use num_traits::{FromPrimitive, Zero};
+use runtime::{ActorCode, Runtime};
+use vm::{
+    ActorError, ExitCode, MethodNum, PieceInfo, Serialized, TokenAmount, METHOD_CONSTRUCTOR,
+    METHOD_SEND,
+};
+
+/// Market actor methods available
+#[derive(FromPrimitive)]
+#[repr(u64)]
+pub enum Method {
+    Constructor = METHOD_CONSTRUCTOR,
+    AddBalance = 2,
+    WithdrawBalance = 3,
+    HandleExpiredDeals = 4,
+    PublishStorageDeals = 5,
+    VerifyDealsOnSectorProveCommit = 6,
+    OnMinerSectorsTerminate = 7,
+    ComputeDataCommitment = 8,
+}
+impl Method {
+    /// Converts a method number into a Method enum
+    fn from_method_num(m: MethodNum) -> Option<Method> {
+        FromPrimitive::from_u64(m)
+    }
+}
+/// Market Actor
+pub struct Actor;
+impl Actor {
+    pub fn constructor<BS, RT>(rt: &mut RT) -> Result<(), ActorError>
+    where
+        BS: BlockStore,
+        RT: Runtime<BS>,
+    {
+        rt.validate_immediate_caller_is(std::iter::once(&*SYSTEM_ACTOR_ADDR))?;
+
+        let empty_root = Amt::<Cid, BS>::new(rt.store()).flush().map_err(|e| {
+            rt.abort(
+                ExitCode::ErrIllegalState,
+                format!("Failed to create market state: {}", e),
+            )
+        })?;
+
+        let empty_map = make_map(rt.store()).flush().map_err(|err| {
+            rt.abort(
+                ExitCode::ErrIllegalState,
+                format!("Failed to create market state: {}", err),
+            )
+        })?;
+
+        let empty_m_set = Multimap::new(rt.store()).root().map_err(|e| {
+            ActorError::new(
+                ExitCode::ErrIllegalState,
+                format!("Failed to create market state: {}", e),
+            )
+        })?;
+
+        let st = State::new(empty_root, empty_map, empty_m_set);
+        rt.create(&st)?;
+        Ok(())
+    }
+    /// Attempt to withdraw the specified amount from the balance held in escrow.
+    /// If less than the specified amount is available, yields the entire available balance.
+    pub fn withdraw_balance<BS, RT>(
+        rt: &mut RT,
+        params: WithdrawBalanceParams,
+    ) -> Result<(), ActorError>
+    where
+        BS: BlockStore,
+        RT: Runtime<BS>,
+    {
+        let (nominal, recipient) = escrow_address(rt, &params.provider_or_client)?;
+
+        let mut amount_slashed_total = TokenAmount::zero();
+        let mut amount_extracted = TokenAmount::zero();
```

Comment:
What does this do? It gets assigned at line 125 and that's it. It never gets used...